cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -84,13 +99,101 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
-                    "image" to "corda/corda-zulu-${it.runtimeVersion().toLowerCase()}"
+                    "image" to (it.dockerImage ?: "corda/corda-zulu-${it.runtimeVersion().toLowerCase()}")
             )
-        }.toMap()
 
+            if (it.dockerConfig.isNotEmpty()) {
+
+                var dockerConfig = ConfigFactory.parseMap(it.dockerConfig)
+
+                val dockerfile = when {
+                    dockerConfig.hasPath("dbDockerConfig.dockerfile") -> dockerConfig.getString("dbDockerConfig.dockerfile")
+                    else -> DEFAULT_DB_DOCKERFILE
+                }
+                val dbInit = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbInit") -> dockerConfig.getString("dbDockerConfig.dbInit")
+                    else -> DEFAULT_DB_INIT_FILE
+                }
+                val dbName = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbName") -> dockerConfig.getString("dbDockerConfig.dbName")
+                    else -> DEFAULT_DB_NAME
+                }
+                val dbSchema = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbSchema") -> dockerConfig.getString("dbDockerConfig.dbSchema")
+                    else -> DEFAULT_DB_SCHEMA
+                }
+                val dbUser = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbUser") -> dockerConfig.getString("dbDockerConfig.dbUser")
+                    else -> DEFAULT_DB_USER
+                }
+                val dbPassword = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbPassword") -> dockerConfig.getString("dbDockerConfig.dbPassword")
+                    else -> DEFAULT_DB_PASSWORD
+                }
+                val dbDataSourceClassName = when {
+                    dockerConfig.hasPath("dataSourceProperties.dataSourceClassName") -> dockerConfig.getString("dataSourceProperties.dataSourceClassName")
+                    else -> DEFAULT_DB_DATA_SOURCE_CLASS_NAME
+                }
+                val dbTransactionIsolationLevel = when {
+                    dockerConfig.hasPath("database.transactionIsolationLevel") -> dockerConfig.getString("database.transactionIsolationLevel")
+                    else -> DEFAULT_DB_TRANSACTION_ISOLATION_LEVEL
+                }
+                val dbRunMigration = when {
+                    dockerConfig.hasPath("database.runMigration") -> dockerConfig.getBoolean("database.runMigration")
+                    else -> DEFAULT_DB_RUN_MIGRATION
+                }
+
+                val dbPort = DEFAULT_DB_STARTING_PORT + index
+                val dbHost = "${it.containerName}-db"
+
+                val dbUrl = when {
+                    dockerConfig.hasPath("dataSourceProperties.dataSource.url") -> {
+                        var url = dockerConfig.getString("dataSourceProperties.dataSource.url")
+                        url.replace("\${DBHOSTNAME}", dbHost)
```

Comment:
The problem now is that we are declaring that we support _exactly_ these four tokens for expansion.